e2e/LongHaul/amqpPooling/DeviceOperations.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Mash.Logging;
+using Microsoft.Azure.Devices.Client;
+using static Microsoft.Azure.Devices.LongHaul.AmqpPooling.LoggingConstants;
+
+namespace Microsoft.Azure.Devices.LongHaul.AmqpPooling
+{
+    internal class DeviceOperations
+    {
+        private readonly IotHubDeviceClient _deviceClient;
+        private readonly string _deviceId;
+        private readonly Logger _logger;
+
+        private long _totalTelemetryMessagesSent = 0;
+
+        public DeviceOperations(IotHubDeviceClient deviceClient, string deviceId, Logger logger)
+        {
+            _deviceClient = deviceClient;
+            _deviceId = deviceId;
+            logger.LoggerContext.Add("deviceId", deviceId);
+            _logger = logger;
+        }
+
+        public IotHubDeviceClient DeviceClient => _deviceClient;
+
+        public async Task<Task> SendAsync(TelemetryMessage message, CancellationToken ct)
+        {
+            try
+            {
+                var sw = new Stopwatch();
+
+                _logger.Trace($"Sending a telemetry message from the device with Id [{_deviceId}].", TraceSeverity.Information);
+                sw.Restart();
+
+                await _deviceClient.SendTelemetryAsync(message, ct).ConfigureAwait(false);
+                sw.Stop();
+
+                _logger.Metric(TelemetryMessageDelaySeconds, sw.Elapsed.TotalSeconds);
+                _logger.Metric(TotalTelemetryMessagesSent, ++_totalTelemetryMessagesSent);
+
+                return Task.CompletedTask;
```

Comment:
```suggestion
                return;
```

Reply:
I tried to return a `Task` from this method which would be added into a `List<Task>` further, so I can call [this](https://github.com/Azure/azure-iot-sdk-csharp/blob/eea3f874c82f893562abf16e9b206fa2ea5ccf86/e2e/LongHaul/amqpPooling/IotHub.cs#L114) to kick off operations for all devices together.

Reply:
Something is wrong with this implementation, though. You shouldn't need to do it.